management/src/main/scala/akka/management/internal/HealthChecksImpl.scala

```
@@ -0,0 +1,90 @@
+  import system.dispatcher
+
+  private val log = Logging(system, classOf[HealthChecksImpl])
+
+  log.info("Loading readiness checks {}", settings.readinessChecks)
+  log.info("Loading liveness checks {}", settings.livenessChecks)
+
+  private val readiness: immutable.Seq[HealthCheck] = load(settings.readinessChecks)
+
+  private val liveness: immutable.Seq[HealthCheck] = load(settings.livenessChecks)
+
+  private def load(
+      checks: immutable.Seq[String]
+  ): immutable.Seq[HealthCheck] = {
+    checks
+      .map(
+        fqcn =>
+          system.dynamicAccess
+            .createInstanceFor[HealthCheck](
+              fqcn,
+              immutable.Seq((classOf[ActorSystem], system))
+            )
+            .recoverWith {
+              case _: ClassCastException =>
+                system.dynamicAccess
+                  .createInstanceFor[java.util.function.Supplier[CompletionStage[Boolean]]](fqcn,
+                    immutable.Seq((classOf[ActorSystem], system)))
+                  .map(sup => () => sup.get().toScala)
+          }
+      )
+      .map {
+        case Success(c) => c
+        case Failure(_: NoSuchMethodException) =>
+          throw new InvalidHealthCheckException(
+            s"Health checks: [${checks.mkString(",")}] must have a single argument constructor that takes an ActorSystem"
+          )
+        case Failure(_: ClassCastException) =>
+          throw new InvalidHealthCheckException(
+            s"Health checks: [${checks.mkString(",")}] must have type: () => Future[Boolean]"
+          )
+        case Failure(c: ClassNotFoundException) =>
+          throw new InvalidHealthCheckException(
+            s"Health check: [${c.getMessage}] not found"
+          )
+        case Failure(t) =>
+          throw new InvalidHealthCheckException(
+            "Uncaught exception from Health check construction",
+            t
+          )
+      }
+  }
+
+  def ready(): Future[Boolean] = {
+    check(readiness)
```

maybe also protect against faulty implementations that throw exception instead of returning failed future

---

:+1: